src/index.ts

```diff
@@ -82,6 +85,53 @@ async function installWrangler() {
 		);
 	}
 
+	startGroup("🔍 Checking for existing Wrangler installation");
+	let installedVersion = "";
+	try {
+		const { stdout } = await getExecOutput(
+			packageManager.exec,
+			["wrangler", "--version"],
+			{
+				cwd: config["workingDirectory"],
+				silent: true,
+			},
+		);
+		const versionMatch = stdout.match(/wrangler (\d+\.\d+\.\d+)/);
+		if (!versionMatch) {
+			throw new Error(
+				`Unable to parse Wrangler version from the output: ${stdout}`,
+			);
+		}
+		installedVersion = versionMatch[1];
+		if (config.didUserProvideWranglerVersion) {
+			if (semverEq(config.WRANGLER_VERSION, installedVersion)) {
+				info(`✅ Using Wrangler ${installedVersion}`, true);
+				endGroup();
+				return;
```

Comment:
This nested return makes it a bit difficult to follow the flow here.
I think it would be cleaner if we use this block just to try identifying a currently installed version.
Then another block to determine whether we should use it or not.
And then finally a block that will install a version of we can't use the current version.